src/codegen/codegen_cuda.cc

```diff
@@ -414,6 +456,9 @@ void CodeGenCUDA::VisitStmt_(const Allocate* op) {
       stream << ' ';
       PrintType(op->type, stream);
     }
+    if ((op->type == Int(4) || op->type == UInt(4) || op->type == Int(1)) && scope == "shared") {
```

Comment:
I'm not sure why we need to check the scope. Can we allocate Int4/Int1 buffer at global or local?